### PR TITLE
feat: add 'script-defer' option for `injectRegister` script

### PR DESCRIPTION
**Pull request by LouisMazel**
To optimize page loading speed

https://www.webpagetest.org/result/231122_BiDcG6_7SP/2/experiments/#Quick

image

### CODE REVIEW

**Comment by userquin** [comment]
@LouisMazel thx, I'll try to review both PR this weekend

File: docs/guide/register-service-worker.md [hunk 1]
```diff
@@ -47,7 +48,7 @@ When configuring `injectRegister: 'inline'` in the plugin configuration, the plu
 
 ## Script Registration
 
-When configuring `injectRegister: 'script'` in the plugin configuration, the plugin will generate a `registerSW.js` script adding it to your application entry point:
+When configuring `injectRegister: 'script'` or `injectRegister: 'script-defer'` in the plugin configuration, the plugin will generate a `registerSW.js` script adding it to your application entry point:
```
Comment (userquin) [inline docs/guide/register-service-worker.md:51]: can you remove any change in the docs here? docs in this repo is deprecated

File: src/types.ts [hunk 2]
```diff
@@ -87,13 +87,15 @@ export interface VitePWAOptions {
    *
    * `inline` - inject a simple register, inlined with the generated html
    *
-   * `script` - inject <script/> in <head>, with the `sr` to a generated simple register
+   * `script` - inject `<script/>` in `<head>`, with the `sr` to a generated simple register
```
Comment (userquin) [inline src/types.ts:90]: replace this with:
`script` - inject `<script/>` in `<head>` with `src` attribute to a generated script to register the service worker

apply same change in L92 below for `script-defer`

**Comment by userquin** [comment]
I will add some changes tmr for dev:
- `src/plugins/dev.ts` should handle also `script-defer`
- review websocket logic

**Comment by userquin** [comment]
@LouisMazel I'm adding all requested changes, you don't need to worry about my requests, I'm changing `strict-defer` to `script-defer`

**Comment by userquin** [comment]
imagen

**Review by userquin — APPROVED** [review]
thx @LouisMazel I'm going to fix another problem here and will review your PR in docs repo

**Comment by LouisMazel** [comment]
Thank you @userquin!

🚀🚀🚀

**Pull request merged**